web/modules/custom/dbcdk_community/src/Plugin/Block/ProfileBlock.php

```diff
@@ -0,0 +1,247 @@
+      '#header' => [
+        $this->t('Subject'),
+        $this->t('Information'),
+      ],
+      '#rows' => (isset($rows) ? $rows : ''),
+      '#empty' => $this->t('There were no information to be found for the profile "%username".', [
+        '%username' => $username,
+      ]),
+      '#cache' => [
+        'max-age' => 0,
+      ],
+    ];
+    // Provide additional information/functionality to the profile table if the
+    // profile exists.
+    if (!empty($profile)) {
+      // Prefix and suffix can't handle a renderable array so we have to
+      // render the array to markup by ourselves.
+      $build['table']['#suffix'] = \Drupal::service('renderer')->render($this->getActionButtons($profile));
+    }
+
+    return $build;
+  }
+
+  /**
+   * Parse a Community Service Profile.
+   *
+   * Takes a Community Service Profile and an array of fields to determine
+   * what values that should be display in the table of profile information.
+   * Most fields will simply have to be returned with their values but some
+   * of the fields require different treatments (either because they should
+   * be presented in another way or it's an object of any kind).
+   *
+   * @param Profile $profile
+   *   A Communtiy Service Profile.
+   * @param array $fields
+   *   An array of fields in the following format: field_name => title.
+   *
+   * @return array $rows
+   *   An array of rows containing subjects and information.
+   */
+  protected function parseProfile(Profile $profile, array $fields) {
+    $rows = [];
+    foreach ($fields as $field => $title) {
+      // Check if the field requires any special treatment or just use the
+      // default behavior and return the value from the field.
+      switch ($field) {
+        // The birthday field returns a DateTime object and should be formatted
+        // with a Drupal Date Format instead.
+        case 'birthday':
+          $date_formatter = \Drupal::service('date.formatter');
+          $value = $date_formatter->format($profile->getBirthday()->getTimestamp(), 'dbcdk_community_service_birthday');
+          break;
+
+        // We have to allow HTML in the output of the description field since
+        // Twig automatically escapes string variables because markup is meant
+        // to be handled in templates.
+        case 'description':
+          $value = [
+            'data' => [
+              '#markup' => $profile->getDescription(),
+              '#allowed_tags' => Xss::getAdminTagList(),
+            ],
+          ];
+          break;
+
+        default:
+          // Use the machine-name of the field to fetch its value from the
+          // Profile object (format: $profile->getFieldName()).
+          $value = $profile->{'get' . ucfirst($field)}();
+
+          // The render array cannot take an object as a column value (this
+          // will cause a fatal error). Instead we check if it is an object
+          // and return an empty string and log the event.
```

Comment:
It will cause a fatal error if the value happened to be i.e. an object since it cannot be converted to a string